django/db/views.py

```diff
@@ -1397,31 +1398,77 @@ def smallMoleculeDetail(request, facility_salt_id):
                 where=where, order_by=('facility_id',))        
             details['datasets'] = DataSetTable(queryset)
         
-        # nominal target dataset results information
+        # Target Affinity Spectrum data
         try:
-            dataset = DataSet.objects.get(dataset_type='Nominal Targets')
-            metaWhereClause=[
-                '''"smallMolecule" ~ '^%s' ''' % sm.facility_salt,
-                '''"isNominal" = '1' '''] 
-            ntable = DataSetManager2(dataset).get_table(
-                metaWhereClause=metaWhereClause,
-                column_exclusion_overrides=['isNominal'])
-            logger.info(str(('ntable',ntable.data, len(ntable.data))))
+            dataset = DataSet.objects.get(facility_id='20000')
```

Comment:
The `try` block only needs to contain this one query in order to issue the warning, right? If the dataset doesn't exist, the new query would just return zero rows. That will lead to an empty targets table but no exceptions. Dataset 20000 isn't going to just disappear by accident, so we don't need to bend over backwards to handle that case.

Reply:
Yes, actually, by the same logic, we don't need to query for the Dataset at all; and the smallMoleculeDetail template checks for the target_affinity_table before rendering that section, so if there are no rows, it can just be excluded.
- I'll remove the extra code here as it is not necessary.